packages/core/src/utilities/isEqual.ts

```diff
@@ -1,27 +1,72 @@
+function areNumbersEqualWithTolerance(
+  num1: number,
+  num2: number,
+  tolerance: number
+): boolean {
+  return Math.abs(num1 - num2) <= tolerance;
+}
+
+function areArraysEqual(
+  arr1: number[] | Float32Array | Float64Array,
+  arr2: number[] | Float32Array | Float64Array,
+  tolerance = 1e-5
+): boolean {
+  if (arr1.length !== arr2.length) {
+    return false;
+  }
+
+  for (let i = 0; i < arr1.length; i++) {
+    if (!areNumbersEqualWithTolerance(arr1[i], arr2[i], tolerance)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function isNumberType(value: any): value is number {
+  return typeof value === 'number';
+}
+
+function isFloatArrayLike(
+  value: any
+): value is number[] | Float32Array | Float64Array {
+  const isArrayLike = typeof value === 'object' || 'length' in value;
+  const floatTypedArrayLike =
+    value instanceof Float32Array || value instanceof Float64Array;
+
+  return isArrayLike || floatTypedArrayLike;
```

Comment:
so this becomes true if we have Int8Array too right? because of `isArrayLike`, then why the function name suggests it should be float array?